<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Gene ID Mapping for Genotype-Tissue Expression (GTEx) Data • grex</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Gene ID Mapping for Genotype-Tissue Expression (GTEx) Data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">grex</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.9.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/grex.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/nanxstats/grex/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Gene ID Mapping for Genotype-Tissue Expression (GTEx) Data</h1>
                        <h4 data-toc-skip class="author">Nan Xiao &lt;<a href="https://nanx.me" class="external-link uri">https://nanx.me</a>&gt;<br> Gao Wang
&lt;<a href="https://www.tigerwang.org" class="external-link uri">https://www.tigerwang.org</a>&gt;<br> Lei Sun &lt;<a href="mailto:sunl@uchicago.edu" class="email">sunl@uchicago.edu</a>&gt;</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/nanxstats/grex/blob/v1.9.1/vignettes/grex.Rmd" class="external-link"><code>vignettes/grex.Rmd</code></a></small>
      <div class="d-none name"><code>grex.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The Genotype-Tissue Expression (GTEx) project <span class="citation">(Lonsdale et al. 2013)</span> aims at measuring human
tissue-specific gene expression levels. With the collected data, we will
be able to explore the landscape of gene expression, gene regulation,
and their deep connections with genetic variations.</p>
<p>Raw GTEx data contains expression measurements from various types of
elements (such as genes, pseudogenes, noncoding DNA sequences) covering
the whole genome. For some analysis, it might be desirable to only keep
a subset of the data, for example, data from protein coding genes. In
such cases, mapping the original Ensembl gene IDs to Entrez gene IDs or
HGNC symbols become an essential step in the analysis pipeline.</p>
<p><code>grex</code> offers a <strong>minimal dependency</strong>
solution to do such ID mappings. Currently, an Ensembl ID from GTEx can
be mapped to its Entrez gene ID, HGNC gene symbol, and UniProt ID, with
basic annotation information such as HGNC gene name, cytogenetic
location, and gene type. We also limit our scope on the Ensembl IDs
appeared in the gene read count data. Ensembl IDs from transcript data
will be considered in future versions.</p>
</div>
<div class="section level2">
<h2 id="mapping-table">Mapping table<a class="anchor" aria-label="anchor" href="#mapping-table"></a>
</h2>
<p>To facilitate such ID conversion tasks, the <code>grex</code> package
has a built-in mapping table derived from the well-known annotation data
package <code>org.Hs.eg.db</code> <span class="citation">(Carlson
2015)</span>. The mapping data we used has integrated mapping
information from Ensembl and NCBI, to maximize the possibility of
finding a matched Entrez ID. The R script for creating the mapping table
is located <a href="https://github.com/nanxstats/grex/blob/master/data-raw/mapping-table-generator.R" class="external-link">here</a>.</p>
<p>Not surprisingly, when creating such a table, there were hundreds of
cases where a single Ensembl ID can be mapped to multiple Entrez gene
IDs. To create a one-to-one mapping, we took a simple approach: we just
removed the duplicated Entrez IDs and only kept the first we encountered
in the original database. Therefore, there might be cases where the
mapping is not 100% accurate. If you have such doubts for particular
results, please try searching the original ID on the Ensembl website and
see if we got a correct mapped ID.</p>
</div>
<div class="section level2">
<h2 id="code-example">Code example<a class="anchor" aria-label="anchor" href="#code-example"></a>
</h2>
<p>As an example, we use the Ensembl IDs from GTEx V7 gene count data
and select 100 IDs:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://nanx.me/grex/">"grex"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"gtexv7"</span><span class="op">)</span></span>
<span><span class="va">id</span> <span class="op">&lt;-</span> <span class="va">gtexv7</span><span class="op">[</span><span class="fl">101</span><span class="op">:</span><span class="fl">200</span><span class="op">]</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/grex.html">grex</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span>
<span><span class="co">#&gt;          ensembl_id entrez_id hgnc_symbol                                   hgnc_name</span></span>
<span><span class="co">#&gt; 95  ENSG00000266075      &lt;NA&gt;        &lt;NA&gt;                                        &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 96  ENSG00000272153      &lt;NA&gt;        &lt;NA&gt;                                        &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 97  ENSG00000116198      9731      CEP104                     centrosomal protein 104</span></span>
<span><span class="co">#&gt; 98  ENSG00000169598      1677        DFFB       DNA fragmentation factor subunit beta</span></span>
<span><span class="co">#&gt; 99  ENSG00000198912    339448    C1orf174         chromosome 1 open reading frame 174</span></span>
<span><span class="co">#&gt; 100 ENSG00000236423 100133612   LINC01134 long intergenic non-protein coding RNA 1134</span></span>
<span><span class="co">#&gt;     cyto_loc uniprot_id   gene_biotype</span></span>
<span><span class="co">#&gt; 95      &lt;NA&gt;       &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 96      &lt;NA&gt;       &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 97   1p36.32     O60308 protein_coding</span></span>
<span><span class="co">#&gt; 98   1p36.32     B4DZS0 protein_coding</span></span>
<span><span class="co">#&gt; 99   1p36.32     Q8IYL3 protein_coding</span></span>
<span><span class="co">#&gt; 100  1p36.32       &lt;NA&gt;        lincRNA</span></span></code></pre></div>
<p>The elements which cannot be mapped accurately will be
<code>NA</code>.</p>
<p>Genes with a mapped Entrez ID:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filtered_genes</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">df</span><span class="op">[</span></span>
<span>    <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="st">"entrez_id"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ensembl_id"</span>, <span class="st">"entrez_id"</span>, <span class="st">"hgnc_symbol"</span>, <span class="st">"gene_biotype"</span><span class="op">)</span></span>
<span>  <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">filtered_genes</span><span class="op">)</span></span>
<span><span class="co">#&gt;        ensembl_id entrez_id hgnc_symbol         gene_biotype</span></span>
<span><span class="co">#&gt; 1 ENSG00000162576     54587       MXRA8       protein_coding</span></span>
<span><span class="co">#&gt; 2 ENSG00000175756     54998    AURKAIP1       protein_coding</span></span>
<span><span class="co">#&gt; 4 ENSG00000221978     81669       CCNL2       protein_coding</span></span>
<span><span class="co">#&gt; 5 ENSG00000224870    148413  MRPL20-AS1 processed_transcript</span></span>
<span><span class="co">#&gt; 6 ENSG00000242485     55052      MRPL20       protein_coding</span></span>
<span><span class="co">#&gt; 8 ENSG00000235098    441869     ANKRD65       protein_coding</span></span></code></pre></div>
<p>If you want to start from the raw GENCODE gene IDs provided by GTEx
(e.g. <code>ENSG00000227232.4</code>), the function
<code><a href="../reference/cleanid.html">cleanid()</a></code> can help you remove the <code>.version</code>
part in them, to produce Ensembl IDs.</p>
</div>
<div class="section level2">
<h2 id="whats-next">What’s next?<a class="anchor" aria-label="anchor" href="#whats-next"></a>
</h2>
<p>Conventionally, the next step is removing (or imputing) the genes
with <code>NA</code> IDs, and then select the genes to keep. Notably, as
was observed in the complete gene read count data, in about 100 cases,
multiple Ensembl IDs can be mapped to one single Entrez ID.
Post-processing steps may also be needed for such genes.</p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>We thank members of the <a href="http://stephenslab.uchicago.edu" class="external-link">Stephens lab</a> (Kushal K Dey,
Michael Turchin) for their valuable suggestions and helpful discussions
on this problem.</p>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-hsdb" class="csl-entry">
Carlson, Marc. 2015. <em><span class="nocase">org.Hs.eg.db</span>:
Genome Wide Annotation for Human</em>.
</div>
<div id="ref-gtex" class="csl-entry">
Lonsdale, John, Jeffrey Thomas, Mike Salvatore, Rebecca Phillips, Edmund
Lo, Saboor Shad, Richard Hasz, et al. 2013. <span>“The Genotype-Tissue
Expression (<span>GTEx</span>) Project.”</span> <em>Nature Genetics</em>
45 (6): 580–85.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Nan Xiao, Gao Wang, Lei Sun.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
